plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2337,6 +2338,9 @@ producer_plugin_impl::handle_push_result( const transaction_metadata_ptr& trx,
          if ( !trx->is_read_only() )
             pr.block_exhausted = block_is_exhausted(); // smaller trx might fit
          pr.trx_exhausted = true;
+      } else if (trace->except->code() == ro_trx_temporary::code_value) {
+         fc_dlog(_trx_failed_trace_log, "[TRX_TRACE] Read-only trx execution temporary failure, tx: ${txid} RETRYING", ("txid", trx->id()));
+         pr.retry = true;
```

Comment:
Instead of this `else if`, I think you should add `ro_trx_temporary::code_value` to `exception_is_exhausted` list and consider this just another exhausted case.

Then the above `fc_dlog(_trx_failed_trace_log` in the `exception_is_exhausted` `if` block should be `fc_dlog( is_transient ? _transient_trx_failed_trace_log : _trx_failed_trace_log`

Reply:
Thanks. This simplifies coding.